docs/source-2.0/spec/selectors.rst

```diff
@@ -1471,6 +1471,36 @@ Implementations MAY choose to evaluate ``:root`` expressions eagerly or
 lazily, though they MUST evaluate ``:root`` expressions no more than once.
 
 
+``:recursive``
+--------------
+
+The ``:recursive`` function applies a selector to the current shape, and for
+every shape yielded that has yet to be yielded, applies the selector to that
```

Comment:
"for every shape yielded that has yet to be yielded" 
This wording is kinda confusing. I wonder if there's a clearer way to say this?